nep-TokenStandard.mediawiki

```diff
@@ -0,0 +1,162 @@
+
+If the method succeeds, it MUST fire the <code>TransferBatch</code> event, and MUST return <code>true</code>, even if the token is sent to the owner.
+
+If the transfer is not processed, the function SHOULD return <code>false</code>.
+
+
+====balanceOf====
+<pre>
+public static BigInteger balanceOf(byte[] account, byte[] tokenId)
+</pre>
+Returns the token balance of the account if the token type is Fungible Token. If the token type is Non-Fungible token, this method SHOULD return <code>1</code> if the account is the owner of this NFT, otherwise it SHOULD return <code>0</code>.
+
+The parameter account SHOULD be a 20-byte address. If not, this method SHOULD throw an exception.
+
+If the account is an unused address, this method MUST return <code>0</code>.
+
+
+====tokensOf====
+<pre>
+public static enumerator tokensOf(byte[] owner)
+</pre>
+Returns an enumerator that contains all of the token ids owned by the specified address.
+
+The parameter owner SHOULD be a 20-byte address. If not, this method SHOULD throw an exception.
+
+====ownerOf====
+<pre>
+public static byte[] ownerOf(byte[] tokenId)
+</pre>
+Returns an account address  that own the specified NFT.
+
+The parameter tokenId SHOULD be a valid NFT. If not, this method SHOULD throw an exception.
+
+
+====properties====
+<pre>
+public static string properties(byte[] tokenId)
+</pre>
+
+Returns a serialized NVM object containing the properties for the given NFT. The NVM object must conform to the [https://github.com/neo-project/proposals/blob/61e53a07a319c537c57c988cbcb216ea75de0f89/nep-11.mediawiki#neo-nft-metadata-json-schema NEO NFT Metadata JSON Schema].
+
+The parameter <code>tokenId</code> SHOULD be a valid index of that NFT type. If not, this method SHOULD throw an exception.
+
+====mintNonFungible====
+<pre>
+public static bool mintNonFungible(byte[] type, byte[] to)
+</pre>
+
+Mint a NFT to an address.
+
+The user who invoke this function MUST be the creator of this type of NFT. If not, this method SHOULD throw an exception.
+
+The invocation of mintNonFungible function SHOULD have at least one <code>NEO</code> as mortgage
+If not, this method SHOULD throw an exception.
+
+The parameter <code>type</code> SHOULD be a valid NFT type.  If not, this method SHOULD throw an exception.
+
+The parameter <code>to</code> SHOULD be a 20-byte address. If not, this method SHOULD throw an exception.
+
+If the method succeeds, it MUST fire the Transfer event, and MUST return true.
+
+If the Mint is not processed, the function SHOULD return <code>false</code>.
+
+====mintFungible====
+<pre>
+public static bool mintFungible(byte[] type, byte[] to,  BigInteger amount)
+</pre>
+Mint an <code>amount</code> of Fungible Token to an address.
+
+The user who invoke this function MUST be the creator of this type of Fungible Token . If not, this method SHOULD throw an exception.
+
```

Comment:
By `type` you mean the `hash` of the contract, @steven1227?

Reply:
Here `type` is the unique mark of one kind NFT/FT. In this standard, the hash of this contract is just the address of this contract, has no use inside the contract. Each kind of NFT/FT in this standard will be created by the creator with an assigned `type`. The `type` can be used to clarify which NFT/FT a token belongs, and also used to distinguish if it is an NFT or FT by using bit masking on this `type`